firebase-vscode/src/data-connect/ad-hoc-mutations.ts

```diff
@@ -162,46 +176,79 @@ query {
-    const mutationDataType: GraphQLInputObjectType = schema.getTypeMap()[
-      dataName
-    ] as GraphQLInputObjectType;
-
-    // build mutation as string
-    const functionSpacing = "\t";
-    const fieldSpacing = "\t\t";
-    const mutation = [];
-    mutation.push("mutation {"); // mutation header
-    mutation.push(`${functionSpacing}${lowerCaseName}_insert(data: {`);
-    for (const [fieldName, field] of Object.entries(
-      mutationDataType.getFields(),
-    )) {
-      // necessary to avoid type error
-      const fieldtype: any = field.type;
-      // use all argument types that are of scalar, except x_expr
-      if (
-        isDataConnectScalarType(fieldtype.name) &&
-        !field.name.includes("_expr")
-      ) {
-        const defaultValue = (defaultScalarValues as any)[fieldtype.name] || "";
-        mutation.push(
-          `${fieldSpacing}${fieldName}: ${defaultValue} # ${fieldtype.name}`,
-        ); // field name + temp value + comment
-      }
+    for (const field of fields) {
+      const type = getNamedType(field.type);
+      const defaultValue = getDefaultScalarValue(type.name);
+      if (!defaultValue) continue;
+
+      argumentFields.push({
+        kind: Kind.OBJECT_FIELD,
+        name: { kind: Kind.NAME, value: field.name },
+        value: defaultValue,
+      });
     }
-    mutation.push(`${functionSpacing}})`, "}"); // closing braces/paren
-    return mutation.join("\n");
+
+    return {
+      kind: Kind.OPERATION_DEFINITION,
+      operation: OperationTypeNode.MUTATION,
+      name: { kind: Kind.NAME, value: singularName },
+      selectionSet: {
+        kind: Kind.SELECTION_SET,
+        selections: [
+          {
+            kind: Kind.FIELD,
+            name: { kind: Kind.NAME, value: `${singularName}_insert` },
+            arguments: [
+              {
+                kind: Kind.ARGUMENT,
+                name: { kind: Kind.NAME, value: "data" },
+                value: {
+                  kind: Kind.OBJECT,
+                  fields: argumentFields,
+                },
+              },
+            ],
+          },
+        ],
+      },
+    };
   }
 
+  function getDefaultScalarValue(type: string): ValueNode | undefined {
+    switch (type) {
+      case "Any":
+        return { kind: Kind.OBJECT, fields: [] };
+      case "Boolean":
+        return { kind: Kind.BOOLEAN, value: false };
+      case "Date":
+        return {
+          kind: Kind.STRING,
+          value: new Date().toISOString().substring(0, 10),
+        };
+      case "Float":
+        return { kind: Kind.FLOAT, value: "0" };
+      case "Int":
+        return { kind: Kind.INT, value: "0" };
+      case "Int64":
+        return { kind: Kind.INT, value: "0" };
+      case "String":
+        return { kind: Kind.STRING, value: "" };
+      case "Timestamp":
+        return { kind: Kind.STRING, value: new Date().toISOString() };
+      case "UUID":
+        return { kind: Kind.STRING, value: "1111-2222-3333-44444" };
```

Comment:
```suggestion
        return { kind: Kind.STRING, value: "11111111222233334444555555555555" };
```

to make it actually parse. Removing dashes to align with the default output format (so people don't copy a UUID and wonder if they should manually insert dashes)

Reply:
counter point... empty string with an LSP error message might be better, to make it clear they need to provide a truly unique ID?

otherwise, if we're keeping a valid ID, I'd stick to something easier to read, like all 0's: "00000000000000000000..."